File: RetailStation.Entities/DTOs/HR/DepartmentModel.cs
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.HR
{
    public class DepartmentModel : CreatorModel
    {
        public int? DepartmentId { get; set; }
        public int? BranchId { get; set; }
        public string BranchNameAR { get; set; }
        public string BranchNameEN { get; set; }
        public string Code { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        public string Location { get; set; }
        public bool? IsSystem { get; set; }
        public string Description { get; set; }
        public int? ManagerId { get; set; }

        public string ManagerNameAR { get; set; }
        public string ManagerNameEN { get; set; }
        public int? TotalCount { get; set; }

    }
}

File: RetailStation.Angular/src/app/components/HR/models/DepartmentModel.ts
import { CreatorModel } from "../../Shared/models/CreatorModel";


export interface DepartmentModel extends CreatorModel {
    departmentId: number | null;
    branchId: number | null;
    branchNameAR: string;
    branchNameEN: string;
    code: string;
    nameAR: string;
    nameEN: string;
    location: string;
    isSystem: boolean | null;
    description: string;
    managerId: number | null;
    managerNameAR: string;
    managerNameEN: string;
}